bip-0183.md

```diff
@@ -0,0 +1,570 @@
+| blockhash                  | 32 byte vector               | The blockhash for this utreexo accumulator state                                                                                                                                                                 |
+| length of the root hashes  | varint                       | The length of the root hashes                                                                                                                                                                                    |
+| root hashes                | vector of 32 byte hashes     | The utreexo roots for the UTXO set at the blockhash. See [roots](bip-0181.md#Definitions)                                                                                                       |
+| length of the proof hashes | varint                       | The length of the proof hashes                                                                                                                                                                                   |
+| proof hashes               | vector of 32 byte hashes     | The proof hashes needed to validate with the pre-committed utreexo accumulator of the utreexo roots                                                                                                              |
+
+This message is for implementing an out-of-order block validation node[^2] or softchains[^3].
+
+Because the size of the state needed to validate blocks is so small with Utreexo, nodes can perform IBD in parallel and out of order.
+
+For example, a computer could divide the task of validating 800,000 blocks into 100 tasks of 8,000 blocks each: blocks 1 through 800, 800 through 1600, 1600 through 2400, and so on.
+
+In order start the 1600 through 2400 IBD task, however, the node should know what the state of the utxo set is at block 1600, so that it can validate and modify the accumulator.
+
+In order to do this, the binary can provide "linkup hints", where the state of the accumulator is given for a desired block hash.
+
+While giving the state of the system might seem at first glance to be introducing a trust assumption, these are not trusted states.
+The node performing IBD tries out the state given for a block height, but checks that when that state is reached from the thread "below" that it properly links up, with the accumulator state arrived at through full validation matching the state given.
+If that link up does not successfully happen, the IBD process should halt.
+
+These hints are statements of fact that are hard-coded into the program itself, and if they are false all bets are off about the program.
+
+Archive nodes create a forest of Linkup hints, so that they can prove, with respect to the Linkup forest roots in a node performing IBD, what their binary has claimed the utxo accumulator state to be at any block height.
```

Comment:
```suggestion
Archive nodes create a forest of Linkup hints, so that they can prove, with respect to the Linkup forest roots in a node performing IBD, what their binary has claimed the UTXO accumulator state to be at any block height.
```